test/test_jit.py

```diff
@@ -9223,6 +9223,58 @@ def fn(lst):
 
         self.checkScript(fn, ([torch.ones(2) + 2, torch.ones(2)],))
 
+    def test_ignore_decorator(self):
+        class M(torch.jit.ScriptModule):
+            def __init__(self):
+                super(M, self).__init__()
+
+            @torch.jit.script_method
+            def forward(self, x):
+                if self.training:
+                    self.ignored_code(x)
+                return x
+
+            @torch.jit.ignore
+            def ignored_code(self, x):
+                return 10
+
+        self.assertNotIn('ignored_code', str(M().graph))
```

Comment:
This is suspicious. Wasn't the whole point of this patch to allow you to run this code in Python, but ignore it only when you actually export the model?

Reply:
Like, you might still want to have a half-optimized training pathway

Reply:
The point was to completely skip over certain functions, no matter where they are executed. I think it's potentially confusing to change the behavior of a model on export.